### PR TITLE
mcs: support forwarding split and scatter request

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Before this PR, the API service was responsible for scatter and split. After this PR, the request will be forwarded to the scheduling service. Once we implement the scheduling service discovery, we can directly send requests to the scheduling service.

```commit-message
```

### Check List

Tests

- Manual test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/scheduling/server/grpc_service.go [hunk 1]
```diff
@@ -191,6 +185,48 @@ func (s *Service) StoreHeartbeat(ctx context.Context, request *schedulingpb.Stor
 	return &schedulingpb.StoreHeartbeatResponse{Header: &schedulingpb.ResponseHeader{ClusterId: s.clusterID}}, nil
 }
 
+// SplitRegions split regions by the given split keys
+func (s *Service) SplitRegions(ctx context.Context, request *schedulingpb.SplitRegionsRequest) (*schedulingpb.SplitRegionsResponse, error) {
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/grpc_service.go:189]: Do we need to add some tests about forward requests?

Comment (rleungx) [reply pkg/mcs/scheduling/server/grpc_service.go:189]: Yes, I have added a TODO in the PR description. Will do it later.

Comment (lhy1024) [inline pkg/mcs/scheduling/server/grpc_service.go:189]: BTW, we seem to need to support `AskSplit`, `AskBatchSplit`, and `GetOperator` with grpc method too.

Comment (rleungx) [reply pkg/mcs/scheduling/server/grpc_service.go:189]: We don't need AskSplit/AskBatchSplit. But GetOperator is necessary.

Comment (lhy1024) [reply pkg/mcs/scheduling/server/grpc_service.go:189]: `HandleAskBatchSplit` will call `GetMergeChecker()`

Comment (rleungx) [reply pkg/mcs/scheduling/server/grpc_service.go:189]: I think we can move the split cache out, WDYT?

Comment (rleungx) [reply pkg/mcs/scheduling/server/grpc_service.go:189]: Oh, you are right.

Comment (rleungx) [reply pkg/mcs/scheduling/server/grpc_service.go:189]: Will do it in the next PR

**Review by JmPotato — APPROVED** [review]
LGTM. Please update the kvproto first.

File: server/grpc_service.go [hunk 2]
```diff
@@ -2175,7 +2283,7 @@ func (s *GrpcServer) SplitRegions(ctx context.Context, request *pdpb.SplitRegion
 }
 
 // SplitAndScatterRegions split regions by the given split keys, and scatter regions.
-// Only regions which splited successfully will be scattered.
+// Only regions which split successfully will be scattered.
 // scatterFinishedPercentage indicates the percentage of successfully splited regions that are scattered.
 func (s *GrpcServer) SplitAndScatterRegions(ctx context.Context, request *pdpb.SplitAndScatterRegionsRequest) (*pdpb.SplitAndScatterRegionsResponse, error) {
```
Comment (lhy1024) [inline server/grpc_service.go:2288]: Do we need to forward this functions?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: server/grpc_service.go (hunk 2 above)
Comment (rleungx) [reply server/grpc_service.go:2288]: AFAIK, this API is not used for now.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**